source/common/secret/secret_manager_impl.h

```diff
@@ -68,6 +71,17 @@ class SecretManagerImpl : public SecretManager {
       return secret_provider;
     }
 
+    std::vector<std::shared_ptr<SecretType>> allSecretProviders() {
+      std::vector<std::shared_ptr<SecretType>> providers;
+      for (const auto& secret_entry : dynamic_secret_providers_) {
```

Comment:
is it safe to iterate ```dynamic_secret_providers_```  when  ```dynamic_secret_providers_[map_key] = secret_provider``` 
 or  ```dynamic_secret_providers_.erase(map_key);``` runs ?

Reply:
I checked last time, both config dump  and the xDS updates callback handler are invoked from main thread, so no sync coordination is required, but i can be wrong.

let's wait for envoy folks to double check this.

Reply:
That's right, admin are handled in main thread so it is safe.

Reply:
thanks for confirming. Resolve this.

commenting/doc about this thread safety does not seem necessary to me, but let me know if that's needed.